doc/source/data/key-concepts.rst

```diff
@@ -146,6 +146,17 @@ scheduling strategy for all Datasets tasks/actors, using the global
   :start-after: __resource_allocation_begin__
   :end-before: __resource_allocation_end__
 
+Fault tolerance
+===============
+
+Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>` in Ray core. Specifically, a ``Dataset`` will be automatically recovered by Ray in case of failures. This works through **lineage reconstruction**: a Dataset is a collection of Ray objects stored in shared memory, and if any of these objects are lost, then Ray will recreate these by re-executing the task(s) that created them.
+
+There are a few cases that are not currently supported:
+1. If the original creator of the ``Dataset`` dies. This is because the ``Dataset creator`` stores the metadata for the :ref:`objects <object-fault-tolerance>` that comprise the ``Dataset``.
```

Comment:
```suggestion
1. If the original creator of the ``Dataset`` dies. This is because the creator stores the metadata for the :ref:`objects <object-fault-tolerance>` that comprise the ``Dataset``.
```